cmake/targets/BuildBun.cmake

```
@@ -846,6 +846,12 @@ else()
     set(LLD_NAME lld-${LLVM_VERSION_MAJOR})
   endif()
 
+  if(ARCH STREQUAL "x86_64")
```

This check always fails.

```suggestion
  if(ARCH STREQUAL "x64")
```